### ISSUE TITLE
Snap.js Modes: Reveal, Push, Over

### ISSUE THREAD

**Issue by prud** [jakiestfu/Snap.js#65]
Current implementation: You drag the content area to the left/right to reveal the drawer that is visible below (in terms of z-depth) the content area.

I'd like to use it this way for our iOS app, however for Android I need a slightly different behavior:
On Android the new pattern suggests that you start dragging on the content area to actually move the revealing navigation from the left. The drawer will be on above the content, which is not moved at all. 

![image](https://f.cloud.github.com/assets/2361882/604760/28ae21d4-ccff-11e2-8b53-aabfbb2eb961.png)

Do you see anything like this being available as an option? To make this easier, I'd suggest to ignore the static Android header for now and keep it full page.

Source: http://developer.android.com/design/patterns/navigation-drawer.html

**Comment by jakiestfu** [comment]
Thanks for providing good detail and imagery.

There are three main styles to pull out the menu/drawer: 
- Reveal (which Snap.js currently uses) 
- Push (which pushes content away at the same time menu slides out)
- Over, which is exactly what you describe above. The menu will slide out over the content as it does in many android apps.

As soon as I have more free time to focus on Snap.js again, I intend to include all three of these styles and it will be as easy as setting a single parameter to utilize them.

**Comment by t0rpedo** [comment]
I agree that this would expand the overall appeal of snap.js. +1 for these options!

**Comment by yumyo** [comment]
that's great! +1

**Comment by prud** [comment]
Another suggestion for a reveal mode would be a different kind of 'Reveal', lets say 'DynamicReveal, which slightly translates the background drawer on the x-axis while dragging. Its used on iOS7 as illustrated here: 
![screencast1](https://f.cloud.github.com/assets/2361882/1122238/1413bbe2-1ad6-11e3-9201-e2b1ea0c7712.jpg)

However, just a feature suggestion, no high priority!

**Comment by jakiestfu** [comment]
@prud Performance is a huge impact on Snap.js, especially when it means firing a custom event everytime the user drags the screen x amount of pixels. Adding a parallax I fear may degrade the quality and efficiency of this code, but it will be something I'd like to experiment with

**Comment by prud** [comment]
Like I said, just an idea I was playing around with and not very important after all. Stay focused on the performance stuff and keep this for future reference or a possible plugin. - sent via Hojoki

**Comment by Dindaleon** [comment]
Hi there, I see the topic is very old and you have not implemented the suggested styles. It would be very nice to have the 'over' style.

**Comment by rttmax** [comment]
would be great +1

**Comment by hkirsman** [comment]
+1 for over style

**Comment by giuliolastra** [comment]
+1 for over style

**Comment by nhh** [comment]
yeah - "move" as option would be pretty nice!
